### PR TITLE
[no ticket][risk=no] e2e simplify code that returns boolean 

### PR DESCRIPTION
**Pull request by aweng98**
Description: Belated follow-up PR that addresses the feedback in [RW-8015][risk=no] new e2e Controlled Tier workspace test (PR #6426).

Turn this (verbose) code
```
> +      .waitForXPath(noSearchResultsXpath, { visible: true, timeout: 100 })
+      .then(() => {
+        return false;
+      })
+      .catch(() => {
+        return true;
+      });
```

to this (simpler) code
`.waitForXPath(noSearchResultsXpath, { visible: true, timeout: 100 }).then(() => false).catch(() => true)
`

Created a new common util function to avoid code duplication.
```
export async function exists(page: Page, xpath: string, opt: { timeout?: number } = {}): Promise<boolean> {
  const { timeout = 100 } = opt;
  return page
    .waitForXPath(xpath, { visible: true, timeout })
    .then(() => false)
    .catch(() => true);
}
```

**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

[RW-8015]: https://precisionmedicineinitiative.atlassian.net/browse/RW-8015?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by aweng98** [comment]
@dmohs Hi, ptal.

File: e2e/utils/element-utils.ts [hunk 1]
```diff
@@ -67,3 +67,11 @@ export async function matchText(page: Page, cssSelector: string, subString: stri
   }
   return found;
 }
+
+export async function exists(page: Page, xpath: string, opt: { timeout?: number } = {}): Promise<boolean> {
```
Comment (calbach) [inline e2e/utils/element-utils.ts:71]: Perhaps we should rename this to `isVisible`; something that matches the semantics of what the underlying Puppeteer check is doing. If we want to use `exists` as the name, then probably we should have `visible` be an option - which could default to true.

File: e2e/utils/runtime-utils.ts [hunk 2]
```diff
@@ -19,12 +20,8 @@ export async function initializeRuntimeIfModalPresented(page: Page): Promise<voi
 }
 
 async function isSecuritySuspended(page: Page): Promise<boolean> {
-  try {
-    await page.waitForXPath('//*[@data-test-id="security-suspended-msg"]', { visible: true });
-  } catch (e) {
-    return false;
-  }
-  return true;
+  const xpath = '//*[@data-test-id="security-suspended-msg"]';
```
Comment (calbach) [inline e2e/utils/runtime-utils.ts:23]: nit: I don't think this separate variable is improving readability, would just inline

File: e2e/app/page/notebook-cell.ts [hunk 3]
```diff
@@ -47,12 +47,8 @@ export default class NotebookCell extends NotebookFrame {
     const selector = this.cellSelector(cellIndx);
     return iframe
       .waitForSelector(`${selector}.selected`, { visible: true })
-      .then(() => {
-        return true;
-      })
-      .catch(() => {
-        return false;
-      });
+      .then(() => true)
```
[the remark this answers is not in the document]
Comment (aweng98) [reply e2e/app/page/notebook-cell.ts]: Yes, updated.

File: e2e/utils/element-utils.ts (hunk 1 above)
Comment (aweng98) [reply e2e/utils/element-utils.ts]: I like the second suggestion. Sometimes we may want to check if an element exists but not visible.
Updated func to
```
export async function exists(
  page: Page,
  xpath: string,
  opt: { visible?: boolean; timeout?: number } = {}
): Promise<boolean> {
  const { visible = true, timeout = 1000 } = opt;
  return page
    .waitForXPath(xpath, { visible, timeout })
    .then(() => true)
    .catch(() => false);
}
```

File: e2e/utils/runtime-utils.ts (hunk 2 above)
Comment (aweng98) [reply e2e/utils/runtime-utils.ts]: Done

**Review by calbach — APPROVED** [review]

**Pull request merged**